### PR TITLE
Docs reorg phase1

### PR DESCRIPTION
**Pull request by ned1313**
In this phase of the docs reorg, there is one new section and one updated section

Contributing - this section will now include a developer contributing guide and a documentation contributing guide. The CONTRIBUTING.md file that was in the Dev directory has been moved.

Reference - this new section will contain reference material including the API, CLI, and Helm chart. Since API and CLI are generated dynamically, we need to update that process to point at the `/reference/api` and `/reference/cli` directories instead of their current path. All references to the API and CLI paths will also need to be updated.

We need to work out the proper order for that update before this gets merged.

### CODE REVIEW

**Comment by ned1313** [comment]
I've updated the relevant Helm, API, and CLI generation code with the new directory target. Now I just need a way to test the full generation and rendering process before we merge to master.

**Comment by kdorosh** [comment]
The following pages look a little off (after a `make serve-site`):
- http://localhost:1313/api/
- http://localhost:1313/cli/
- http://localhost:1313/reference/cli/ (as the content has moved)

Also should API/CLI go under reference instead of at the top level?

File: docs/content/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/virtual_service.proto.sk.md [hunk 1]
```diff
@@ -180,7 +180,7 @@ When a request matches on a route, the route can perform one of the following ac
 - *Route* the request to a destination
 - Reply with a *Direct Response*
 - Send a *Redirect* response to the client
-- *Delegate* the action for the request to one or more top-level [`RouteTable`]({{< ref "/api/github.com/solo-io/gloo/projects/gateway/api/v1/route_table.proto.sk.md" >}}) resources
+- *Delegate* the action for the request to one or more top-level [`RouteTable`]({{< ref "/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/route_table.proto.sk.md" >}}) resources
```
Comment (kdorosh) [inline docs/content/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/virtual_service.proto.sk.md]: let's use `versioned_link_path`

File: docs/content/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/virtual_service.proto.sk.md [hunk 2]
```diff
@@ -151,7 +151,7 @@ An HTTP request is first matched to a virtual host based on its host header, the
 
 If a request is not matched to any virtual host or a route therein, the target proxy will reply with a 404.
 
-Unlike the [Gloo Virtual Host]({{< ref "/api/github.com/solo-io/gloo/projects/gloo/api/v1/proxy.proto.sk.md" >}}/#virtualhost),
+Unlike the [Gloo Virtual Host]({{< ref "/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/proxy.proto.sk.md" >}}/#virtualhost),
```
Comment (kdorosh) [inline docs/content/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/virtual_service.proto.sk.md]: let's use `versioned_link_path`

**Pull request merged**